includes/create-theme/resolver_additions.php

```diff
@@ -106,6 +110,25 @@ public static function get_user_data() {
 				: parent::get_user_data();
 		}
 
+		/**
+		 * This checks if the fontSizes has been set and saved to the database.
+		 * If it has then the $data variable is changed and will later be used to save back the theme.
+		 * A method is used from the parent class if it exists to get any user changed settings.
+		 *
+		 * @param array $data
+		 * @return array $data
+		 */
+		public static function font_size_preset_changes( $data ) {
+			$user_data = parent::get_user_data();
+			if ( method_exists( $user_data, 'get_settings' ) ) {
+				$user_data = $user_data->get_settings();
+				if ( isset( $user_data['typography']['fontSizes'] ) && ! empty( $user_data['typography']['fontSizes'] ) ) {
+					$data['settings']['typography']['defaultFontSizes'] = false;
+					$data['settings']['typography']['fontSizes']        = $user_data['typography']['fontSizes']['default'];
+				}
+			}
+			return $data;
+		}
 		/**
```

Comment:
the function that gets and user data and then uses the parent class method if it exists to access data that is private, then checks if fontSizes is set then appends the settings needs to write to the theme.json file.